bus-mapping/src/circuit_input_builder/transaction.rs

```diff
@@ -486,67 +485,70 @@ impl Transaction {
 /// Transaction L1 fee for L1GasPriceOracle contract
 #[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
 pub struct TxL1Fee {
+    /// chain id
+    pub chain_id: u64,
+    /// block number
+    pub block_number: u64,
     /// L1 base fee
     pub base_fee: u64,
     /// L1 fee overhead
     pub fee_overhead: u64,
     /// L1 fee scalar
     pub fee_scalar: u64,
-    #[cfg(feature = "l1_fee_curie")]
     /// L1 blob fee
     pub l1_blob_basefee: u64,
-    #[cfg(feature = "l1_fee_curie")]
     /// L1 commit scalar
     pub commit_scalar: u64,
-    #[cfg(feature = "l1_fee_curie")]
     /// l1 blob scalar
     pub blob_scalar: u64,
 }
 
 impl TxL1Fee {
     /// Calculate L1 fee and remainder of transaction.
-    /// for non curie upgrade case, tx_rlp_signed_len is not used,  set to zero
-    pub fn tx_l1_fee(&self, _tx_data_gas_cost: u64, tx_rlp_signed_len: u64) -> (u64, u64) {
-        // <https://github.com/scroll-tech/go-ethereum/blob/49192260a177f1b63fc5ea3b872fb904f396260c/rollup/fees/rollup_fee.go#L118>
-        #[cfg(not(feature = "l1_fee_curie"))]
-        {
-            let tx_l1_gas = _tx_data_gas_cost + self.fee_overhead + TX_L1_COMMIT_EXTRA_COST;
-            let tx_l1_fee = self.fee_scalar as u128 * self.base_fee as u128 * tx_l1_gas as u128;
-            (
-                (tx_l1_fee / TX_L1_FEE_PRECISION as u128) as u64,
-                (tx_l1_fee % TX_L1_FEE_PRECISION as u128) as u64,
-            )
+    pub fn tx_l1_fee(&self, tx_data_gas_cost: u64, tx_rlp_signed_len: u64) -> (u64, u64) {
+        let is_curie = is_curie_enabled(self.chain_id, self.block_number);
+        if is_curie {
```

Comment:
`before_curie` and `after_curie` are revered order ?
   if is_curie {
            self.tx_l1_fee_after_curie(tx_data_gas_cost)
        } else {
            self.tx_l1_fee_before_curie(tx_rlp_signed_len)

Reply:
yes... thanks..